app/controllers/ForumPost.scala

```diff
@@ -37,30 +38,44 @@ final class ForumPost(env: Env) extends LilaController(env) with ForumController
         if topic.closed then BadRequest("This topic is closed")
         else if topic.isOld then BadRequest("This topic is archived")
         else
-          categ.team.so(env.team.api.isLeader(_, me)) flatMap { inOwnTeam =>
-            forms
-              .post(inOwnTeam)
-              .bindFromRequest()
-              .fold(
-                err =>
-                  CategGrantWrite(categId, tryingToPostAsMod = true):
-                    for
-                      captcha     <- forms.anyCaptcha
-                      unsub       <- env.timeline.status(s"forum:${topic.id}")
-                      canModCateg <- access.isGrantedMod(categ.slug)
-                      page <- renderPage:
-                        html.forum.topic
-                          .show(categ, topic, posts, Some(err -> captcha), unsub, canModCateg = canModCateg)
-                    yield BadRequest(page)
-                ,
-                data =>
-                  CategGrantWrite(categId, tryingToPostAsMod = ~data.modIcon):
-                    CreateRateLimit(ctx.ip, rateLimited):
-                      postApi.makePost(categ, topic, data) map { post =>
-                        Redirect(routes.ForumPost.redirect(post.id))
-                      }
-              )
-          }
+          for
+            canModCateg <- access.isGrantedMod(categ.slug)
+            replyBlocked <- access.isReplyBlockedOnUBlog(topic, canModCateg)
```

Comment:
Since, you are currently making changes to this plz change this to `ctx.me soUse access.isReplyBlockedOnUBlog(topic, canModCateg)`

Reply:
Is that not needed here?

Reply:
it's not, because there's a given `Me` instance in the scope.

```scala
def create(categId: ForumCategId, slug: String, page: Int) = AuthBody { ctx ?=> me ?=>
```

Reply:
in `show` there may or may not be a context user.
But in `create`, we know for sure there is one: anons can't post.